src/Promise/PromiseInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+namespace GraphQL\Promise;
+
+interface PromiseInterface
+{
+    /**
+     * Waits until the promise completes if possible.
+     *
+     * @return mixed
+     * @throws \LogicException if the promise has no wait function.
+     */
+    public function wait();
+
```

Comment:
`wait` method is equivalent to `await` in node implementation (with [react](https://github.com/mcg-web/promise-factory/blob/master/src/Factory/ReactPromiseFactory.php#L86-L110) vs with [guzzle](https://github.com/mcg-web/promise-factory/blob/master/src/Factory/GuzzleHttpPromiseFactory.php#L91-L120)) . this helps to get promise completion without parsing by `then`. Minimum interface to promise should be like [that](https://github.com/mcg-web/promise-factory/blob/master/src/PromiseFactoryInterface.php). I try a lot different promise implementation in past weeks. I can provide a full working version (with all tests) using my lib. No more need of using PromiseInterface or PromiseWrapper with this lib. I can understand that using a external requirements it not really what we want here, but promise is a more and more a global need for a lot of projects, for example thats what we use in dataloader lib. We can think of a PromiseAdapter but in a separate project, to be easily shared...